distributed/worker.py

```diff
@@ -3543,7 +3627,31 @@ def get_call_stack(self, comm=None, keys=None):
     def _notify_plugins(self, method_name, *args, **kwargs):
         for name, plugin in self.plugins.items():
             if hasattr(plugin, method_name):
-                if method_name == "release_key":
+                if (
+                    not getattr(plugin, "enum_task_state_names", False)
```

Comment:
```suggestion
                    not plugin.enum_task_state_names
```
I think getattr is unnecessary - aren't all customer plugins supposed to subclass WorkerPlugin?

Reply:
Historically I don't think we've enforced this. That's why, for example, we have `hasattr` checks like this 

https://github.com/dask/distributed/blob/1a2313ce63ccbb7e0d69ddb9471c84a0c9bb53cf/distributed/worker.py#L1539-L1543

Enforcing plugins to subclass `WorkerPlugin` doesn't sound like a bad idea, but would merit a deprecation cycle